x-pack/plugins/security/server/saved_objects/saved_objects_security_extension.ts

```diff
@@ -218,6 +389,49 @@ export class SavedObjectsSecurityExtension implements ISavedObjectsSecurityExten
       throw this.errors.decorateGeneralError(error, error.body && error.body.reason);
     }
   }
+
+  async authorizeCreate(params: {
```

Comment:
This is a working example of what an action authorization method would look like. It handles both  the create and bulk create actions depending on the count of the objects passed in.

Reply:
> It handles both the create and bulk create actions depending on the count of the objects passed in.

Do consumers need to explicitly specify `action` as a parameter then? Also would that logic be the same in case `bulkGet` is invoked with just one object?

Reply:
So, I had rolled the action parameter back in just so I could satisfy some failing unit tests (should have updated my comment). The issue was that the bulk operations expect a bulk action (passed into the lower level methods, and in the audit for example). If we want to maintain this (for traceability?) we need a way to differentiate non-bulk actions from bulk action with 1 object. Not sure how often this occurs, but it certainly is possible and does happen.
So, in more recent changes (not pushed yet), I replaces the action parameter with an options parameter that contains a 'forceBulkAction' flag. Not exactly graceful, but allows for the differentiation.

Reply:
I've updated to the paradigm of having bulk and non-bulk public methods, and an internal worker method. This really makes usage clear and concise but still leverages a single logic chain so we don't have unnecessary duplicate code.